checker/jdk/nullness/src/java/util/EnumMap.java

```diff
@@ -0,0 +1,839 @@
+        throw new RuntimeException("skeleton method");
+        // Does not compile with actual version, java.util.AbstractMap is reduced to skeleton methods
+        // Collection<V> vs = values;
+        // if (vs != null)
+        //     return vs;
+        // else
+        //     return values = new Values();
+    }
+
+    private class Values extends AbstractCollection<V> {
+        public Iterator<V> iterator() {
+            return new ValueIterator();
+        }
+        @Pure public int size() {
+            return size;
+        }
+        @Pure public boolean contains(@Nullable Object o) {
+            return containsValue(o);
+        }
+        public boolean remove(@Nullable Object o) {
+            o = maskNull(o);
+
+            for (int i = 0; i < vals.length; i++) {
+                if (o.equals(vals[i])) {
+                    vals[i] = null;
+                    size--;
+                    return true;
+                }
+            }
+            return false;
+        }
+        public void clear() {
+            EnumMap.this.clear();
+        }
+    }
+
+    /**
+     * Returns a {@link Set} view of the mappings contained in this map.
+     * The returned set obeys the general contract outlined in
+     * {@link Map#keySet()}.  The set's iterator will return the
+     * mappings in the order their keys appear in map, which is their
+     * natural order (the order in which the enum constants are declared).
+     *
+     * @return a set view of the mappings contained in this enum map
+     */
+    public Set<Map.Entry<K,V>> entrySet() {
+        Set<Map.Entry<K,V>> es = entrySet;
+        if (es != null)
+            return es;
+        else
+            return entrySet = new EntrySet();
+    }
+
+    private class EntrySet extends AbstractSet<Map.Entry<K,V>> {
+        public Iterator<Map.Entry<K,V>> iterator() {
+            return new EntryIterator();
+        }
+
+        @Pure public boolean contains(@Nullable Object o) {
+            if (!(o instanceof Map.Entry))
+                return false;
+            Map.Entry<?,?> entry = (Map.Entry<?,?>)o;
+            return containsMapping(entry.getKey(), entry.getValue());
+        }
+        public boolean remove(@Nullable Object o) {
+            if (!(o instanceof Map.Entry))
+                return false;
+            Map.Entry<?,?> entry = (Map.Entry<?,?>)o;
+            return removeMapping(entry.getKey(), entry.getValue());
+        }
+        @Pure public int size() {
+            return size;
+        }
+        public void clear() {
+            EnumMap.this.clear();
+        }
+        public Object[] toArray() {
+            return fillEntryArray(new Object[size]);
+        }
+        @SuppressWarnings({
+            "unchecked",
+            "nullness:argument.type.incompatible", // 'a' is known to be of array class type
+            "nullness:override.param.invalid" // Annotation for to Array are technically incorrect. Refer
```

Comment:
Does the checker use these incorrect annotations, or a more precise version as mentioned in the note in Collection.java?

Minor: change "to Array" to "toArray", for clarity.

Reply:
As per the note for the toArray method in Collection.java, the checker uses some hard-coded rules for defining its behaviour. I'm assuming that Checker will use the same hard-coded rules for this method as well as. (Although I have made an effort to express as much as of its behaviour as possible using annotations). Please correct me if I'm wrong.

Thanks!!

Reply:
1. Please check rather than assuming.
2. Please explicitly say that the checker uses hard-coded rules. That will be clearer than just calling them "technically incorrect", which confused me.  My request is really for clearer documentation.
Thanks!